subscription-service/src/test/kotlin/com/egm/stellio/subscription/config/ScheduledNotificationTest.kt

```diff
@@ -0,0 +1,241 @@
+
+    @MockkBean
+    private lateinit var subscriptionEventService: SubscriptionEventService
+
+    @MockkBean
+    private lateinit var fcmService: FCMService
+
+    private lateinit var wireMockServer: WireMockServer
+
+    @BeforeAll
+    fun beforeAll() {
+        wireMockServer = WireMockServer(WireMockConfiguration.wireMockConfig().port(8089))
+        wireMockServer.start()
+        // If not using the default port, we need to instruct explicitly the client (quite redundant)
+        WireMock.configureFor(8089)
+    }
+
+    @AfterEach
+    fun resetWiremock() {
+        WireMock.reset()
+    }
+
+    @AfterAll
+    fun afterAll() {
+        wireMockServer.stop()
+    }
+
+    /**
+     * Test for function 'extracParam'
+     * We simulate a subscription with several 'entitiesInfo'
+     *
+     * @return list of string
+     */
+
+    @Test
+    fun `it should return path with subscription param`() {
+        val entities = setOf(
+            EntityInfo(
+                id = "urn:ngsi-ld:FishContainment:1234567890".toUri(),
+                idPattern = null,
+                type = "FishContainment"
+            ),
+            EntityInfo(
+                id = null,
+                idPattern = null,
+                type = "FishContainment"
+            ),
+            EntityInfo(
+                id = null,
+                idPattern = ".*FishContainment.*",
+                type = "FishContainment"
+            ),
+            EntityInfo(
+                id = "urn:ngsi-ld:FishContainment:1234567890".toUri(),
+                idPattern = ".*FishContainment.*",
+                type = "FishContainment"
+            )
+        )
+        val q = "speed>50;foodName==dietary fibres"
+
+        val param = scheduledNotification.extractParam(entities, q)
+
+        assertEquals(4, param.size)
+        assertEquals(
+            "/urn:ngsi-ld:FishContainment:1234567890?q=speed>50;foodName==dietary fibres",
+            param.get(0)
+        )
+        assertEquals("?type=FishContainment&q=speed>50;foodName==dietary fibres", param.get(1))
+        assertEquals(
+            "?type=FishContainment&idPattern=.*FishContainment.*&q=speed>50;foodName==dietary fibres",
+            param.get(2)
+        )
+        assertEquals(
+            "/urn:ngsi-ld:FishContainment:1234567890?q=speed>50;foodName==dietary fibres",
+            param.get(3)
+        )
+    }
+
+    /**
+     * Test for function 'getEntity'
+     * We call entity service and we want one entity
+     *
+     * @return JsonLdEntity
+     */
+    @Test
+    fun `it should return one entity`() {
+        WireMock.stubFor(
+            WireMock.get(WireMock.urlMatching("/ngsi-ld/v1/entities/urn:ngsi-ld:BeeHive:TESTC"))
+                .willReturn(WireMock.okJson(loadSampleData("beehive.jsonld")))
+        )
+
+        runBlocking {
+            val entity = scheduledNotification.getEntity("/urn:ngsi-ld:BeeHive:TESTC")
```

Comment:
a bit strange to have this leading `/` for entity id

Reply:
It's because when I extractParam I put `/` when I'ts `id` and `?` the other times